teps/0008-support-knative-service-for-triggers-eventlistener-pod.md

```diff
@@ -0,0 +1,286 @@
+kubernetes deployment with default values and yaml looks something like below.
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+```
+
+#### Knative Service `OR any CRD`
+To support Knative Service along with Kubernetes Deployment we use dynamic `Raw` data so that it can be any CRD like `serving.knative.dev.` 
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+ service:
+   dynamic:
+     apiVersion: serving.knative.dev/v1  #It can be any CRD (foo.bar.com)
+     kind: Service
+     metadata:
+       labels:
+         serving.knative.dev/visibility: "cluster-local"
+     spec:
+       template:
+         metadata:
+           annotations:
+             autoscaling.knative.dev/minScale: "1"
+         spec:
+           serviceAccountName: tekton-triggers-github-sa
+           nodeSelector:
+             app: test
+           tolerations:
+           - key: key
+             value: value
+             operator: Equal
+             effect: NoSchedule
+```
+
+### Validation
+
+Below are the few basic high level validation 
+1. If no `service` field specified as part of `EventListener` the Kubernetes Deployment will be created with default values 
+like the existing behavior because now `serviceAccountName` is optional and if not provided `default` serviceaccount 
+will be used which is tracked by this [issue](https://github.com/tektoncd/triggers/issues/682)
+
+2. If `service` is provided then at a time it can have either `typed` or `dynamic` not both.
+
+3. Validation of all the podSpec and containerSpec fields.
+    * If user provided podSpec and containerSpec fields are not supported triggers webhook can give an error like below
+    ```
+   admission webhook "validation.webhook.triggers.tekton.dev" denied the request: 
+   validation failed: must not set the field(s): spec.template.spec.containers[0].image
+    ```
+    * Should not allow more than one container.
+    
+## Advantages
+
+* Triggers eventlistener now gets the serverless feature by default.
+
+* Along with Knative, triggers eventlistener support any CRD which satisfy the contract.
+
+* No management of dependencies ex: Knative.
+
+## Test Plan
+
+* e2e and unit tests
```

Comment:
Would this require us to run and install knative/serving in our e2e test cluster environments?

Reply:
@dlorenc we might indeed. We should try to install the lightest possible knative (aka only serving, and most likely without istio — either using kourier or nothing)

Reply:
Yes i agree with @vdemeester :+1:  
+1 on the lightest  knative serving installation with kourier as ingress gateway